src/Validator.php

```diff
@@ -1,26 +1,33 @@
 <?php
 
+declare(strict_types=1);
 
 namespace Yiisoft\Validator;
 
 use Yiisoft\Validator\Rule\Callback;
+use Yiisoft\I18n\TranslatorInterface;
 
 /**
  * Validator validates {@link DataSetInterface} against rules set for data set attributes.
  */
 class Validator
 {
+    private ?TranslatorInterface $translator = null;
+    private string $domain;
+
     /**
      * @var Rules[]
      */
     private array $attributeRules = [];
 
-    /**
-     * Validator constructor.
-     * @param $rules
-     */
-    public function __construct(iterable $rules = [])
-    {
+    public function __construct(
+        iterable $rules = [],
+        TranslatorInterface $translator = null,
```

Comment:
I don't think that would work since there is valid usage that doesn't involve validator instance: https://github.com/yiisoft/validator#general-usage